### ISSUE TITLE
Add Support for DateOnly type

### ISSUE THREAD

**Issue by abhishekkumams** [Azure/data-api-builder#1610]
Currently, GraphQL response is showing DateTimeFormat for date type columns.

If the value stored in the DB is `11-03-2017`, graphql will show `11-03-2017T00:00:00Z`. 

![image](https://github.com/Azure/data-api-builder/assets/102276754/5ad8c1e1-fcb3-4e50-86c8-f0b2cca24954)
![image](https://github.com/Azure/data-api-builder/assets/102276754/5656a966-4664-44f4-aa3f-7305e21860b6)

**Comment by abhishekkumams** [comment]
@Aniruddh25, should we do this as well in the next milestone?

**Comment by Aniruddh25** [comment]
@abhishekkumams, makes sense - I wonder why is that the case isn't Date type mapped to HotChocolate Date? https://chillicream.com/docs/hotchocolate/v13/defining-a-schema/scalars#net-scalars

**Comment by abhishekkumams** [comment]
> @abhishekkumams, makes sense - I wonder why is that the case isn't Date type mapped to HotChocolate Date? https://chillicream.com/docs/hotchocolate/v13/defining-a-schema/scalars#net-scalars

Nope, It's mapped to DateTime

**Comment by Aniruddh25** [comment]
@abhishekkumams,

I went back into the previous changes and found this reasoning in the commit: https://github.com/Azure/data-api-builder/commit/63d5ef24e3bb90057fed015caa1d81f222de232a: 

> * DateTime
> 	This is a bit of a tricky one. `DateTime` type is handled by HotChocolate as `DateTimeOffset`. Also, our db column type detection method gives `DateTime` for all `Date`, `DateTime`, and `DateTimeOffset`. Meaning that while the user can define non `DateTimeOffset` columns, they will be interpreted as `DateTimeOffset` and the gaps will be filled with 00:00:00 time and local offset.
> 	HotChocolate also provides us with `Date` type. However, because all db column types `Date`, `DateTime`, and `DateTimeOffset` are detected as `DateTime` system types, we cannot make use of the HotChocolate `Date` type as of now.

So, to solve this, we would need to look for alternative ways for db column type to system type detection method other than ADO.NET Db Data provider.

**Comment by seantleonard** [comment]
@ayush3797 is this duplicate or secondary to your date work?

**Comment by ayush3797** [comment]
> @ayush3797 is this duplicate or secondary to your date work?

Its not.

**Comment by seantleonard** [comment]
@abhishekkumams  i wonder if your work enabling SqlDbType SqlParameter work would help here. When reading the db schema from MsSql, are we able to parse out a specific SqlDbType? or are we still restricted to the limitations of DbType values as explained in Ani's quote

> DateTime https://github.com/Azure/data-api-builder/commit/63d5ef24e3bb90057fed015caa1d81f222de232a
This is a bit of a tricky one. DateTime type is handled by HotChocolate as DateTimeOffset. Also, our db column type detection method gives DateTime for all Date, DateTime, and DateTimeOffset. Meaning that while the user can define non DateTimeOffset columns, they will be interpreted as DateTimeOffset and the gaps will be filled with 00:00:00 time and local offset.
HotChocolate also provides us with Date type. However, because all db column types Date, DateTime, and DateTimeOffset are detected as DateTime system types, we cannot make use of the HotChocolate Date type as of now.

**Comment by JerryNixon** [comment]
## As of March 2026

| Area | Status |
|---|---|
| Input parsing (`DateOnly.Parse`) | ✅ Done |
| `date_only_table` test entity/schema | ✅ Done |
| `SqlDbType.Date → typeof(DateOnly)` mapping | ❌ Still `DateTime` |
| GraphQL scalar for `date` (no time component) | ❌ Still uses `DateTime` scalar |
| `SchemaConverter` handling of `DateOnly` | ❌ Missing case |
| `TypeHelper` JSON/DB type maps for `DateOnly` | ❌ Missing |
